### PR TITLE
Add Quick Open dialog

### PR DESCRIPTION
**Pull request by mitchcurtis**
This dialog searches every Resource-derived file in the project, making
it extremely useful for searching everything from audio files to scripts.

Unlike the Quick Open Script and Quick Open Scene dialogs, you don't need
to think about the type of file you're opening, just its name.

Command + K was chosen because it used for both Qt Creator's locator feature
and Slack's quick switcher feature.

### CODE REVIEW

**Comment by mitchcurtis** [comment]
GIF

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -5694,6 +5700,7 @@ EditorNode::EditorNode() {
 	p->add_separator();
 	p->add_submenu_item(TTR("Open Recent"), "RecentScenes", FILE_OPEN_RECENT);
 	p->add_separator();
+	p->add_shortcut(ED_SHORTCUT("editor/quick_open", TTR("Quick Open..."), KEY_MASK_CMD + KEY_K), FILE_QUICK_OPEN);
```
Comment (akien-mga) [inline editor/editor_node.cpp]: Seems good to me, but I'd make the shortcut Shift+Alt+O to match the existing shortcuts for Quick Open Scene and Quick Open Script.

**Comment by mitchcurtis** [comment]
Done.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**